### PR TITLE
update build system and merge bounded_dispersal

### PR DESCRIPTION
**Pull request by EstebanAce**
Here I updated the build system to use 'pyproject.toml', which is compatible with latest pip/python versions. Also I merged the bounded dispersal branch. Fixed some emergent issues after the merge and added actions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,26 @@
+[![test build](https://github.com/EstebanAce/adascape/actions/workflows/test.yml/badge.svg?branch=master)](https://github.com/EstebanAce/adascape/actions)
```
Comment (benbovy) [inline README.md:1]:
```suggestion
[![test build](https://github.com/fastscape-lem/adascape/actions/workflows/test.yml/badge.svg?branch=master)](https://github.com/fastscape-lem/adascape/actions)
```

(same suggestion for the "test notebooks" badge)

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+]
+[tool.setuptools.packages.find]
+include = [
+    "adascape",
+    "adascape.*",
+]
+
+[tool.setuptools_scm]
+fallback_version = "9999"
+version_file = "adascape/_version.py"
```
Comment (benbovy) [inline pyproject.toml:15]: Not sure version file is really needed.

**Review by benbovy — COMMENTED** [review]
Thanks @EstebanAce ! I left two minor comments for configuration (I haven't reviewed the domain-specific changes made here, I trust you :-) ).

I've changed the repository settings so when you push new commits the github workflows will run hopefully.

**Comment by EstebanAce** [comment]
I added you suggestions. Thanks Benoît!

**Pull request merged**